flink-tests/src/test/java/org/apache/flink/test/checkpointing/SavepointFormatITCase.java

```diff
@@ -111,14 +111,35 @@
                                 keyedState ->
                                         assertThat(
                                                 keyedState,
-                                                instanceOf(KeyGroupsStateHandle.class))));
+                                                instanceOf(KeyGroupsStateHandle.class))),
+                Arguments.of(
+                        SavepointFormatType.CANONICAL,
+                        ROCKSDB_INCREMENTAL_SNAPSHOTS,
+                        (Consumer<KeyedStateHandle>)
+                                keyedState ->
+                                        assertThat(
+                                                keyedState,
+                                                instanceOf(SavepointKeyedStateHandle.class))),
+                Arguments.of(
+                        SavepointFormatType.NATIVE,
+                        ROCKSDB_INCREMENTAL_SNAPSHOTS,
+                        (Consumer<KeyedStateHandle>)
+                                keyedState ->
+                                        assertThat(
+                                                keyedState,
+                                                instanceOf(
+                                                        IncrementalRemoteKeyedStateHandle.class))));
```

Comment:
Is this assertion enough to make sure that native savepoints are working?

Reply:
It is enough to verify the format of the savepoint. We verify savepoints are working if we restore from the savepoint without exceptions.